src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,54 @@
+package mate.academy.service
+
+import mate.academy.model.Circle
+import mate.academy.model.Figure
+import mate.academy.model.FigureEnum
+import mate.academy.model.IsoscelesTrapezoid
+import mate.academy.model.Rectangle
+import mate.academy.model.RightTriangle
+import mate.academy.model.Square
+import java.math.BigDecimal
+import java.math.RoundingMode
+import kotlin.random.Random
+
+private const val WHITE = "white"
+
+private const val DEFAULT_RADIUS = 10.0
+
+private const val DEFAULT_RANGE = 100.0
+
+private const val DEFAULT_SCALE = 2
+
+class FigureSupplier {
+    val colorSupplier: ColorSupplier = ColorSupplier()
+    fun getRandomFigure(): Figure {
+        val randomFigure : FigureEnum
+        = FigureEnum.values()[Random.nextInt(FigureEnum.values().size)]
+        val firstRandomNumber = BigDecimal(Random.nextDouble(DEFAULT_RANGE))
+            .setScale(DEFAULT_SCALE, RoundingMode.HALF_EVEN).toDouble()
+        val secondRandomNumber = BigDecimal(Random.nextDouble(DEFAULT_RANGE))
+            .setScale(DEFAULT_SCALE, RoundingMode.HALF_EVEN).toDouble()
+
+        return when (randomFigure) {
+            FigureEnum.CIRCLE -> Circle(colorSupplier.getRandomColor().toString().lowercase(),
```

Comment:
FigureEnum may be imported for improving readability.
ALT+ENTER -> import FigureEnum